### PR TITLE
Auth: Remove OP permissions on user/table drop

### PR DESCRIPTION
**Pull request by nirbhayc**
Also bump up global auth generation when a permission is granted or revoked.

Signed-off-by: Nirbhay Choubey <nchoubey@bloomberg.net>

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Error &#x26A0;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>3/421 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_truncate_lockorder_generated [core dumped]<br>phys_rep<br>basic_nolibevent_generated<br>sc_truncate_lockorder_generated</p>

File: bdb/bdb_access.c [hunk 1]
```diff
@@ -276,6 +276,15 @@ static int bdb_del_user_tbl_access(bdb_state_type *bdb_state, tran_type *tran,
     int rc = 0;
     int bdberr;
 
+     if ((rc = bdb_tbl_op_access_delete(bdb_state, tran, 0, table_name,
```
Comment (chands10) [inline bdb/bdb_access.c:279]: Looks good to me! Think the layout is one too many spaces here than from the rest of the file though

File: db/bpfunc.c [hunk 2]
```diff
@@ -372,10 +379,17 @@ static int exec_password(void *tran, bpfunc_t *func, struct errstat *err)
     rc = pwd->disable ? bdb_user_password_delete(tran, pwd->user)
                       : bdb_user_password_set(tran, pwd->user, pwd->password);
 
-    if (rc == 0 && pwd->disable) {
-        /* Also delete all the table accesses for this user. */
-        rc = bdb_del_all_user_access(thedb->bdb_env, tran, pwd->user);
-    }
+     if (rc == 0 && pwd->disable) {
```
Comment (chands10) [inline db/bpfunc.c:382]: Also one space extra compared to the rest of the code here

**Review by riverszhang89 — DISMISSED** [review]

File: bdb/bdb_access.c (hunk 1 above)
Comment (nirbhayc) [reply bdb/bdb_access.c]: Fixed, thanks.

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>2/421 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>phys_rep<br>upsert</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>1/421 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>logfill</p>

**Review by riverszhang89 — APPROVED** [review]

**Comment by nirbhayc** [comment]
Thanks @riverszhang89 !!

**Pull request merged**